include/mxnet/engine.h

```diff
@@ -191,19 +208,23 @@ class MXNET_API Engine {
    *                   mutate.
    * \param mutable_vars The variables that current operation will mutate.
    * \param prop Property of the function.
+   * \param need_tracer Whether to record this operation.
    * \param priority Priority of the action, as hint to the engine.
    * \tparam SyncFn the synchronous function to be pushed.
    */
   template<typename SyncFn>
-  inline void PushSync(SyncFn exec_fn, Context exec_ctx,
+  inline void PushSync(std::string const& opr_name,
```

Comment:
put op name as const char*, optional with default, and leave as in last argument,